src/entities/inputValidator/inputValidator.ts

```diff
@@ -28,7 +28,7 @@ export class InputValidator {
     }
 
     validateAddLiquidity(
-        addLiquidityInput: AddLiquidityInput | InitPoolInput,
+        addLiquidityInput: AddLiquidityInput | InitPoolInputV2,
```

Comment:
I see a few errors within `validateTokensAddLiquidity` that are caused by having AddLiquidity and InitPool sharing the validator.
How about splitting them? You could also add a validator to the V3 input, which I think is missing.

Reply:
Good call! Indeed, a validator only for the initPool functionality is more appropriate, done!